### PR TITLE
Test Quarkus S3 extension with Camel Quarkus AWS 2 S3

### LINKED ISSUE #2590: Test camel-quarkus-aws2-s3 together with quarkus-amazon-s3
If the end user adds quarkus-amazon-s3 to an app together with camel-quarkus-aws2-s3 the app should work even if he has to configure them separately. We should add a test for this. There is some potential to clash also in case of other AWS services supported by Quarkus https://github.com/quarkusio/quarkus/tree/main/extensions/amazon-services Maybe in the end the simplest solution would be to make our extensions depend on those from Quarkus. In that way it would not be possible to use them separately and there would be no need for separate tests.

### PR DESCRIPTION
**Pull request by zhfeng**
Fixes #2590 and replace #3107

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -743,6 +743,7 @@
                                         <sortModulesPath>integration-tests/pom.xml</sortModulesPath>
                                         <sortModulesPath>integration-test-groups/pom.xml</sortModulesPath>
                                         <sortModulesPath>integration-test-groups/aws2/pom.xml</sortModulesPath>
+                                        <sortModulesPath>integration-test-groups/aws2-quarkus-client/pom.xml</sortModulesPath>
```
Comment (jamesnetherton) [inline pom.xml:746]: Might be best to remove these changes as I opened a PR to specifically fix up the missing stuff https://github.com/apache/camel-quarkus/pull/3136.

**Review by aldettinger — APPROVED** [review]
Beyond other comments, it looks good to me :+1: Well done @zhfeng

File: pom.xml (hunk 1 above)
Comment (zhfeng) [reply pom.xml:746]: OK, I will do.

**Comment by zhfeng** [comment]
I'll rebase this PR while #3136 get merged.

File: docs/modules/ROOT/pages/reference/extensions/aws2-s3.adoc [hunk 2]
```diff
@@ -44,3 +44,28 @@ Check the xref:user-guide/index.adoc[User guide] for more information about writ
...
+
+== Additional Camel Quarkus configuration
+
+=== Optional integration with Quarkus Amazon S3
+
+If desired, it is possible to use the Quarkus Amazon S3 Client extension in conjunction with Camel Quarkus AWS2 S3.
+Note that this optional and not at all mandatory.
+Follow the https://quarkus.io/guides/amazon-s3#configuring-s3-clients[documentation] with some caveats.
+
+1. The Apache client type has to be selected by configuring the following property.
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/reference/extensions/aws2-s3.adoc:56]: FYI - I had to fix up the list continuation for aws2-ddb in https://github.com/apache/camel-quarkus/pull/3138. The same applies here.

**Review by jamesnetherton — APPROVED** [review]

**Pull request merged**